src/static_file_handler.cc

```diff
@@ -69,18 +69,36 @@ RequestHandler::Status StaticFileHandler::Init(const std::string& uri_prefix, co
             }
         } else if (stmt->tokens_[0] == "username") {
             // Set the username value.
-            username = stmt->tokens_[1];
+            if (username.empty()) {
```

Comment:
Do we want to be able to handle multiple users? Like if we specified in the config the username and password as a pair ("root", "password"). Then have a list of usernames/passwords? It would introduce more complexity though, so it's up to you.